components/Authentication/signup.jsx

```diff
@@ -60,46 +59,90 @@ const ShowText = styled.Text`
   padding-right: 15px;
 `;
 
+const ErrorText = styled.Text`
+  font-family: 'Montserrat_600SemiBold';
+  color: red;
+  font-size: 12px;
+`;
+
+const ViewFiller = styled.View`
+  height: 15px;
+`;
+
 const Signup = ({ navigation }) => {
   const [name, setName] = React.useState('');
+  const [validName, setValidName] = React.useState(false);
   const [email, setEmail] = React.useState('');
+  const [validEmail, setValidEmail] = React.useState(false);
   const [password, setPassword] = React.useState('');
+  const [validPassword, setValidPassword] = React.useState(false);
   const [hidePassword, setHidePassword] = React.useState(true);
+  const isFirstRunName = React.useRef(true);
+  const isFirstRunEmail = React.useRef(true);
+  const isFirstRunPassword = React.useRef(true);
+
+  // TODO: this regex isn't fully functional
+  function ValidateEmail(mail) {
+    // eslint-disable-next-line
+    const re = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    if (re.test(mail)) {
+      return (true);
+    }
+    return (false);
+  }
+
+  React.useEffect(() => {
+    if (isFirstRunName.current) {
+      isFirstRunName.current = false;
+      return;
+    }
+    setValidName(!((name.length > 1)));
+  }, [name]);
+
+  React.useEffect(() => {
+    if (isFirstRunEmail.current) {
+      isFirstRunEmail.current = false;
+      return;
+    }
+    setValidEmail(!ValidateEmail(email));
+  }, [email]);
+
+  React.useEffect(() => {
+    if (isFirstRunPassword.current) {
+      isFirstRunPassword.current = false;
+      return;
+    }
+    setValidPassword(!(password.length > 5));
+  }, [password]);
```

Comment:
Could you explain what you're trying to do with this code, I'm a bit confused by it

Reply:
isFirstRun... tells if it is the first time running, since they're connected to useEffect it runs in the beginning when there is no text in the input. So it makes the error messages only show if it isn't the first time running

Reply:
Oops forgot about forgot password. I'll put that in for login rq.